src/transformers/models/idefics/image_processing_idefics.py

```diff
@@ -0,0 +1,163 @@
+            images (`ImageInput`):
+                Image to preprocess.
+            image_size (`int`, *optional*, defaults to `self.image_size`):
+                Controls the size of the image after `resize`.
+            image_mean (`float` or `List[float]`, *optional*, defaults to `self.image_mean`):
+                Image mean to normalize the image by if `do_normalize` is set to `True`.
+            image_std (`float` or `List[float]`, *optional*, defaults to `self.image_std`):
+                Image standard deviation to normalize the image by if `do_normalize` is set to `True`.
+            eval_mode (`bool`, *optional*, defaults to `self.eval_mode`):
+                Whether to prepare the image for evaluation or training. If `eval_mode` is `True` resize will be
+                performed, otherwise a `RandomResizeCrop`
+            return_tensors (`str` or `TensorType`, *optional*):
+                The type of tensors to return. Can be one of:
+                    - Unset: Return a list of `np.ndarray`.
+                    - `TensorType.TENSORFLOW` or `'tf'`: Return a batch of type `tf.Tensor`.
+                    - `TensorType.PYTORCH` or `'pt'`: Return a batch of type `torch.Tensor`.
+                    - `TensorType.NUMPY` or `'np'`: Return a batch of type `np.ndarray`.
+                    - `TensorType.JAX` or `'jax'`: Return a batch of type `jax.numpy.ndarray`.
+            data_format (`ChannelDimension` or `str`, *optional*, defaults to `ChannelDimension.FIRST`):
+                The channel dimension format for the output image. Can be one of:
+                    - `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                    - `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+        """
+        image_size = image_size if image_size is not None else self.image_size
+        image_mean = image_mean if image_mean is not None else self.image_mean
+        image_std = image_std if image_std is not None else self.image_std
+
+        images = make_list_of_images(images)
+
+        if not valid_images(images):
+            raise ValueError(
+                "Invalid image type. Must be of type PIL.Image.Image, numpy.ndarray, "
+                "torch.Tensor, tf.Tensor or jax.ndarray."
+            )
+
+        # XXX: ideally the transformers set should be in __init__ but then `__repr__` can't handle Compose or Sequential
+        # TypeError: Object of type Sequential is not JSON serializable
+        # alternatively have to override `__repr__` here to do something about it
+        interpolation = transforms.InterpolationMode.BICUBIC
+        if eval_mode:
+            resize_transform = transforms.Resize((self.image_size, self.image_size), interpolation=interpolation)
+        else:
+            resize_transform = transforms.RandomResizedCrop(
+                (self.image_size, self.image_size), scale=(0.9, 1.0), interpolation=interpolation
+            )
```

Comment:
This is the job of the final user. Image processor in Transformer only implement the base processing for inference.